homeassistant/components/evohome/__init__.py

```diff
@@ -248,13 +171,26 @@ async def load_auth_tokens(store) -> tuple[dict, dict | None]:
         session=async_get_clientsession(hass),
     )
 
+    coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_interval=config[DOMAIN][CONF_SCAN_INTERVAL],
+    )
+
     hass.data[DOMAIN] = {}
     hass.data[DOMAIN]["broker"] = broker = EvoBroker(
-        hass, client_v2, client_v1, store, config[DOMAIN]
+        hass, client_v2, client_v1, store, coordinator, config[DOMAIN]
     )
 
-    await broker.save_auth_tokens()
-    await broker.async_update()  # get initial state
+    coordinator.update_method = broker.async_update
+    try:
+        await coordinator.async_config_entry_first_refresh()
+    except ConfigEntryNotReady as exc:
+        # evohome doesn't yet use config flow, so ConfigEntryNotReady is not appropriate
+        ex = RuntimeError()
+        ex.__cause__ = coordinator.last_exception
+        raise ex from exc
```

Comment:
I'm unaware of anything that watches `async_setup` and reacts to a `RuntimeError`. If you want the integration to fail here, simply log `exc` and return False.

Reply:
A very sensible idea! 

However, it got me thinking, and I have now realised I don't need an initial `async_config_entry_first_refresh()` at all, so I have simply removed this block altogether.